### PR TITLE
Add test cases with Rspec

### PR DESCRIPTION
**Pull request by flpfar**
This branch adds tests for Board, BoarChecker and BoardFormatter classes.
In Board, it tests:
- If it initializes the board correctly with two players
- If it validates the moves and inserts them into the board

In BoardChecker:
- Try all the test cases for diagonal, horizontal and vertical to check if there is a winner.
- Returns false if there is no winner.

In BoardFormatter, it checks:
- If the string returned is formatted as expected.

### CODE REVIEW

File: spec/board_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+require_relative '../lib/board'
+
+describe Board do
+  let(:board) { Board.new('Player1', 'Player2') }
+
+  context 'when creating a new board' do
+    it 'creates a board with two players' do
+      expect(board).to be_a(Board)
+      expect(board.players).to eq(0 => 'Player1', 1 => 'Player2')
```
Comment (mosesogwo) [inline spec/board_spec.rb]: Split this into two _**it**_ blocks

File: spec/boardchecker_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+require_relative '../lib/boardchecker'
+
+describe BoardChecker do
+  it 'finds a winner in horizontal positions (Game positions: 123, 456, 789)' do
+    case123 = ['x', 'x', 'x', 4, 'o', 'o', 7, 8, 9]
+    case456 = [1, 'o', 'o', 'x', 'x', 'x', 7, 8, 9]
+    case789 = [1, 'o', 'o', 4, 5, 6, 'x', 'x', 'x']
+    expect(BoardChecker.winner?(case123)).to be true
+    expect(BoardChecker.winner?(case456)).to be true
+    expect(BoardChecker.winner?(case789)).to be true
```
Comment (mosesogwo) [inline spec/boardchecker_spec.rb]: Split this and other such similar setups into separate _**it**_ blocks - one for each expectation.

File: spec/boardchecker_spec.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  it 'finds a winner in vertical positions (Game positions: 147, 258, 369)' do
+    case147 = ['x', 2, 3, 'x', 'o', 'o', 'x', 8, 9]
+    case258 = [1, 'o', 'x', 'x', 'o', 'x', 7, 'o', 9]
+    case369 = [1, 'o', 'x', 4, 5, 'x', 7, 'o', 'x']
+    expect(BoardChecker.winner?(case147)).to be true
+    expect(BoardChecker.winner?(case258)).to be true
+    expect(BoardChecker.winner?(case369)).to be true
```
Comment (mosesogwo) [inline spec/boardchecker_spec.rb]: See previous comments about it blocks.

File: spec/boardchecker_spec.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  it 'finds a winner in diagonal positions (Game positions: 159, 357)' do
+    case159 = ['o', 'x', 'x', 4, 'o', 6, 'x', 8, 'o']
+    case357 = ['o', 2, 'x', 4, 'x', 6, 'x', 8, 'o']
+    expect(BoardChecker.winner?(case159)).to eq(true)
+    expect(BoardChecker.winner?(case357)).to eq(true)
```
Comment (mosesogwo) [inline spec/boardchecker_spec.rb]: See previous comments about it blocks.

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hi @flpfar, @FrancoRosa :clap:. Good work on your tests. Just a few required changes :fast_forward:**
 - You should have one **it** block for each expected behaviour. See comments on your code.


- Add instruction on how to run tests to your README.

_**Kindly make these changes and submit another code review request.**_

**Any question? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

**Comment by flpfar** [comment]
Thank you for reviewing our code, @mosesogwo! The changes in the code were implemented :+1:
About the README, we already have those instructions. Should it be even more descriptive?

**Review by euqueme — CHANGES_REQUESTED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @flpfar and @FrancoRosa :handshake: 

### Thank you for doing the changes requested by the previous reviewer, I think your test cases meet the standards and there's only one more thing to work on before moving on:

#### Project Specific

* [x] specs cover positive and negative scenarios ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/commonsense.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

- I see all your test cases are positive so please add some negative ones using the not_to method and having negative It blocks

Example:

this positive test case could be changed to be a negative test case like this:

```ruby
#Positive
    it 'returns false when the position of the board is already filled' do
      board.register_move(0, 1)
      expect(board.register_move(1, 1)).to be false
    end

#Negative
    it "doesn't return true if the position of the board is already filled" do
      board.register_move(0, 1)
      expect(board.register_move(1, 1)).not_to be true
    end
```

### That's it, Team! :raised_hands: 

### Please :pray: do the changes :man_technologist: :man_technologist: and submit for a code review when you are done :memo: 

Happy coding! :rocket: 

Any questions? ask me on slack **@ Maru**

PROJECT STATUS: REQUESTED CHANGES :traffic_light:

**Comment by flpfar** [comment]
Thank you for reviewing our code, @euqueme ! The requested changes have already been implemented. :+1:

**Review by euqueme — APPROVED** [review]
> ### Pull Request under the review by Maria Eugenia Queme P. (TSE)

### Hello @flpfar and @FrancoRosa :handshake: 

### Thank you for doing the changes, amazing work you may move on! :tada: :tada: :tada: 

PROJECT STATUS: APPROVED :heavy_check_mark:

**Pull request merged**